### ISSUE TITLE
@linaria/shaker process node_modules library error, it is socket.io-client library in my project

### ISSUE THREAD

**Issue by justwalk** [callstack/linaria#1310]
## Environment

- Linaria version: 4.5.4
- "@linaria/babel-preset": "^4.5.4",
    "@linaria/core": "^4.5.4",
    "@linaria/extractor": "^4.5.0",
    "@linaria/logger": "^4.5.0",
    "@linaria/preeval": "^3.0.0-beta.23",
    "@linaria/react": "^4.5.4",
    "@linaria/shaker": "^4.5.3",
    "@linaria/utils": "^4.5.3",
    "@linaria/vite": "^4.5.4",
- vite(+ version): 4.4.7
- Node.js version: 18..0.0
- OS: mac os

## Description
I upgrade my linaria version from "3.0.0-beta.15" to "4.5.0". linaria reported an error, that is 
`Failed to transpile /pages/index/Counter.tsx because:
Cannot read properties of undefined (reading 'fd') in
| /demo/vite-ssr-cli/node_modules/socket.io-client/lib/url.js
| /demo/vite-ssr-cli/node_modules/socket.io-client/lib/index.js` 

my project depend on "socket.io-client" library,  The error occurs when linaria handles the "socket.io-client"

i found linaria shaker would process all dependencies include node_modules. but  sometimes it would be interrupted in some library. here is  "socket.io-client" 

how can i fix it，thanks

## Reproducible Demo
Reproduce is very easy. it just has these code, it would be bugs
`import { Manager } from "socket.io-client";
const socket = new Manager('ws://example.com')
socket.on("disconnect", (reason) => {
  console.log(reason)
});
const style = css`
  color: red;
``

[demo on sandbox](https://codesandbox.io/p/sandbox/prod-meadow-ksk5x9?file=%2Fsrc%2FApp.tsx%3A11%2C1)
https://codesandbox.io/p/sandbox/prod-meadow-ksk5x9?file=%2Fsrc%2FApp.tsx%3A11%2C1

**Comment by layershifter** [comment]
Repro with v5: https://stackblitz.com/edit/vitejs-vite-qsh3gp

## `yarn dev`

```
[plugin:linaria] ENOENT: no such file or directory, open '/home/projects/vitejs-vite-qsh3gp/node_modules/.vite/deps/socket__io-client.js'
```

Looks that there is a problem with resolver.

## `yarn build`

```
error during build:
EvalError: Cannot read properties of undefined (reading 'indexOf') in/home/projects/vitejs-vite-qsh3gp/src/App.jsx
    at Module.evaluate (file:///home/projects/vitejs-vite-qsh3gp/node_modules/@linaria/babel-preset/lib/module.js:224:13)
    at evaluate (file:///home/projects/vitejs-vite-qsh3gp/node_modules/@linaria/babel-preset/lib/evaluators/index.js:15:5)
```

Looks like a problem with shaker, should be debugged.

**Comment by layershifter** [comment]
> yarn dev

This problem is covered and described in #1246.

**Comment by Anber** [comment]
`build` has been fixed in #1354 and #1349. `dev` will be fixed in #1246
